### ISSUE TITLE
jw-nextup-close button lacks aria tags

### ISSUE THREAD

**Issue by ablwr** [jwplayer/jwplayer#1587]
### Expected Behavior

I would like to see the jw-nextup-close button in jwplayer pass an accessibility testing framework as beautifully as its other buttons.

### Actual Behavior

Nested within the `jw-nextup-container` class is a button with the class of `jw-nextup-close`. It seems like this is something that cannot currently be disabled. However, this button lacks aria-label, aria-hidden status, or role (whatever is most applicable in this case).

### Steps to reproduce

Look at any embedded jwplayer through an accessibility framework. [Here is an example sent through the WAVE accessibility report tool](http://wave.webaim.org/report#/https://support.jwplayer.com/customer/portal/articles/1406723#fndtn-method-1). You may have to select "No styles" to the left of the report to see all of the jwplayer parts because the button-in-question is a hidden button.

I'll take a screencap as a shortcut: <img width="730" alt="example of jw-nextup-close button" src="https://cloud.githubusercontent.com/assets/3260492/20988171/a5f7b2c2-bc9d-11e6-9003-c0cb5b933f9b.png">

You can see that it does not have what the other buttons have.

### Environment

Using Mac OSX, Chrome (but I think this is an issue regardless of these factors). JW Player 7.8.2 was tested -- link is above!

### Extra

Thanks so much for otherwise being very compliant and very accessible! Apologies if this is already a known-issue; did a search but didn't see anything related.

**Comment by egreaves** [comment]
Hey @ablwr, thanks for catching this issue and thank you for your kind words. 

This was an oversight on our part as we do strive to make the player as accessible as possible. We'll get this prioritized and addressed asap. 

Alternatively, if you'd like to contribute a PR with a fix, I'd be more than happy to help point you in the right direction.

**Comment by ablwr** [comment]
I would very gladly do that! I trust there is hearty contribution guidelines but happy to take any pointers.

**Comment by ablwr** [comment]
It *seems*, as an outsider looking in gently, that the button-in-question, which lives in `src/templates/nextup.html`, is not something that requires functionality because the "next up" video, when available, populates and can be tabbed-over-to within other divs in the player area and a button for "playing next video" never appears. Clicking this button (when I force it to appear through removing styling) doesn't do anything, either. 

So it seems adding aria-hidden="true" to this button will suffice since it is not meant to be used by a person, just the computer.

However! I could be wrong about this.

Overall though, if this is true, I'm not sure why the HTML element for button is used for this "next up" feature and why it is different from the others control display icons in that it has to differ and use <button> instead of <div> with the button role assigned (like in `src/templates/displayicon.html`)! 

Let me know what I'm not seeing, since I am only looking at the tip of the iceberg.

**Comment by johnBartos** [comment]
Just chiming in here: we want to put `role=button`, `tabindex=0` and an `aria-label` on each, I believe. This is what we we do in our Related overlay

**Comment by ablwr** [comment]
+1 for replacing button element with an empowered div to be in line with the style of all other navigational elements.

**Comment by egreaves** [comment]
The close button is used to close the next up tooltip when it is "sticky" (showing until the end of the video). By default, this happens 10s before the video ends and you can change this by setting `nextupoffset` in the config. Other times during playback, it's not visible and _nextup_ acts as a tooltip that shows when you hover over the _next_ button.

Along with John's recommendations, there are 2 other files that'll need small changes.

We'll need a localization property for the button in `jwplayer/src/js/api/config.js`:
```
localization: {
    play: 'Play',
    ...
    nextUp: 'Next Up',
    related: 'Related'
},
```
We also need to set the `aria-label` attribute of the `jw-nextup-close` button in the setup function in `jwplayer/src/js/view/components/`:

```
        setup: function() {
            this.container = document.createElement('div');
            this.container.className = 'jw-nextup-container jw-reset';
            var element = utils.createElement(nextUpTemplate());
            this.addContent(element);

            this.closeButton = this.content.querySelector('.jw-nextup-close');
            ...
        }
```

Setting it this way allows users of the player to override the defaults by setting their own localized properties in the `localization` block of the config.

**Comment by ablwr** [comment]
This seems straightforward. Am I missing something regarding `jwplayer/src/js/api/config.js`? It seems like the nextUp localization is already declared. Also, since the template is not extended to anything else, the `aria-label` could be hardcoded-asserted in the template. What's the benefit of declaring it in `jwplayer/src/js/view/components`?

I pushed what would seem to be the modest changes [here](https://github.com/ablwr/jwplayer/tree/enhancement/nextup-aria-tags)

The real question is... I compiled jwplayer locally, but how do I test this locally?

**Comment by egreaves** [comment]
Hey @ablwr, thanks again for working on this. I took a look at your changes thus far and we're really close to having this resolved.

You are correct in that there's a `nextUp` localization for the overlay. We could get away with just hard coding `aria-label="Close"`, but having something like `nextUpClose: 'Next Up Close'` in the config specifically for the close button makes it possible to localize the text. 

I'd advise against switching the markup to using a [div for a button](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role). There's wider support for buttons in older user agents and assistive technology. Buttons also support keyboard and focus requirements by default, so we wouldn't need a `tabindex`.

That's great that you've got it compiled. You're 2 steps away from testing it:

1. Add `jwplayer.js` to an HTML page. 
2. Setup the player with a playlist containing more than one item ([reference](https://developer.jwplayer.com/jw-player/docs/developer-guide/customization/configuration-reference/#playlist)). 

By default, the "Next Up" overlay will show with the close button 10s before your media ends, but you can change this by setting `nextupoffset` in your config:

`nextupoffset: 5, // start displaying nextUp 5s after playback starts`
or...
`nextupoffset: -5, // start displaying nextUp 5s before playback ends`

I noticed that you added classes to the close button. The only class needed is `jw-nextup-close` - the player takes care of the rest.

**Comment by ablwr** [comment]
😎! I got this started on my personal computer so I'll take this advice and apply changes post-work, then send a PR.
